tests/integration/test_linspace.py

```diff
@@ -15,46 +15,277 @@
 
 import numpy as np
 import pytest
+from utils.generators import broadcasts_to, mk_seq_array
 
 import cunumeric as num
 
 
-def test_basic():
-    x = np.linspace(2.0, 3.0, num=5)
-    y = num.linspace(2.0, 3.0, num=5)
-    assert np.array_equal(x, y)
+def equivalent_shapes_gen(shape):
+    yield shape
+    for i in range(len(shape) - 1):
+        if shape[i] == 1:
+            i += 1
+            yield shape[i:]
+        else:
+            break
+
+
+@pytest.mark.parametrize(
+    "retstep", (True, False), ids=lambda retstep: f"(retstep={retstep})"
+)
+@pytest.mark.parametrize(
+    "endpoint", (True, False), ids=lambda endpoint: f"(endpoint={endpoint})"
+)
+@pytest.mark.parametrize(
+    "number", (0, 1, 10), ids=lambda number: f"(num={number})"
+)
+@pytest.mark.parametrize(
+    "values",
+    ((10, -5.5), (2.0, 3.0), (0, 0), (1 + 2.5j, 10 + 5j), (0j, 10)),
+    ids=lambda values: f"(values={values})",
+)
+def test_scalar_basic(values, number, endpoint, retstep):
+    start, stop = values
+    x = np.linspace(
+        start, stop, num=number, endpoint=endpoint, retstep=retstep
+    )
+    y = num.linspace(
+        start, stop, num=number, endpoint=endpoint, retstep=retstep
+    )
+
+    if retstep:
+        assert np.array_equal(x[0], y[0])
+        if np.isnan(x[1]) and np.isnan(y[1]):
+            pass
+        else:
+            assert x[1] == y[1]
+    else:
+        assert np.array_equal(x, y)
+
+
+@pytest.mark.parametrize(
+    "retstep", (True, False), ids=lambda retstep: f"(retstep={retstep})"
+)
+@pytest.mark.parametrize(
+    "endpoint", (True, False), ids=lambda endpoint: f"(endpoint={endpoint})"
+)
+def test_arrays_basic(endpoint, retstep):
+    shape = (2, 2, 3)
+    np_start = mk_seq_array(np, shape)
+    num_start = mk_seq_array(num, shape)
+    np_stop = mk_seq_array(np, shape) + 10
+    num_stop = mk_seq_array(num, shape) + 10
+    x = np.linspace(
+        np_start, np_stop, num=5, endpoint=endpoint, retstep=retstep
+    )
+    y = np.linspace(
+        num_start, num_stop, num=5, endpoint=endpoint, retstep=retstep
+    )
+    if retstep:
+        assert np.array_equal(x[0], y[0])
+        assert np.array_equal(x[1], y[1])
+    else:
+        assert np.array_equal(x, y)
+
+
+def test_arrays1():
+    shape_start = (2, 2, 3)
+    np_start = mk_seq_array(np, shape_start)
+    num_start = mk_seq_array(num, shape_start)
+
+    for s in broadcasts_to(shape_start):
+        for shape_stop in equivalent_shapes_gen(s):
```

Comment:
Any reason to to `parametrize` these? 
```
shape_start = (2, 2, 3)
shape_stops = (equivalent_shapes_gen(s) for s in broadcasts_to(shape_start))

@pytest.mark.parametrize("shape_stop", shape_stops)
def test_array_broadcast_stops
```

(maybe there is a better test name)

Reply:
We cannot parameterize in this way.
`shape_stops = (equivalent_shapes_gen(s) for s in broadcasts_to(shape_start))`
It doesn't work as equivalent_shapes_gen(s) returns more than one value.

Reply:
I think you could use [`itertools.chain`](https://docs.python.org/3/library/itertools.html#itertools.chain) then?

Reply:
Fixed.